views/default/page/components/tabs.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Tabbed module component
+ * Provides support for inline and ajax tabbing
+ *
+ * @uses $vars['id']      Optional ID of the module
+ * @uses $vars['class']   Additional classes
+ * @uses $vars['module']  Module name
+ *                        Defaults to 'tabs' (.elgg-module-tabs)
+ * @uses $vars['tabs']    An array of tabs suitable for 'navigation/tabs' view
+ *                        Each tab should specify either:
+ *                        - a 'href' parameter to load content via AJAX, or
+ *                        - a 'content' parameter to display content inline
+ *                        If the tab uses a 'href' attribute, it should specify
+ *                        whether the contents should be reloaded on a subsequent
+ *                        click via 'data-ajax-reload' parameter.
+ *                        You can pass additional data to the ajax view via
+ *                        data-ajax-query attribute (json encoded string).
+ *                        You can also set the data-ajax-href parameter of the tab,
+ *                        which will override the href parameter, in case you want
+ *                        to ensure the tab is clickable event before the JS is bootstrapped.
+ *                        <code>
+ *                        [
+ *                           ['text' => 'Tab 1', 'href' => '/dynamic', 'data-ajax-reload' => true],
+ *                           ['text' => 'Tab 2', 'href' => '/static', 'data-ajax-reload' => false],
+ *                           ['text' => 'Tab 3', 'href' => '/static', 'data-ajax-reload' => false, 'data-ajax-query' => json_encode($data)],
+ *                           ['text' => 'Tab 3', 'href' => '/page', 'data-ajax-href' => '/ajax/page'],
+ *                           ['text' => 'Tab 4', 'href' => 'Tab content'],
+ *                        ]
+ *                        </code>
+ */
+$id = elgg_extract('id', $vars);
+if (!isset($vars['id'])) {
+	$id = "elgg-tabs-" . base_convert(mt_rand(), 10, 36);
+}
+$vars['id'] = $id;
+
+$vars['class'] = elgg_extract_class($vars, 'elgg-tabs-component');
+
+$tabs = (array) elgg_extract('tabs', $vars, []);
+unset($vars['tabs']);
+if (empty($tabs)) {
+	return;
+}
+
+$content = '';
+foreach ($tabs as $index => $tab) {
+	$selected = elgg_extract('selected', $tab);
+
+	if (isset($tab['content'])) {
+		$class = ['elgg-content'];
+		$class[] = $selected ? 'elgg-state-active' : 'hidden';
+
+		$content .= elgg_format_element('div', [
+			'class' => $class,
+			'id' => "{$id}-{$index}",
+		], $tab['content']);
+		unset($tab['content']);
+
+		$tab['href'] = "#{$id}-{$index}";
+	}
```

Comment:
If `content` not given, `href` should be required.
If `href` given, `data-ajax-reload` should be required.
log error if these aren't met.